### PR TITLE
feat(text): allow to pass intl values as well

### PR DESCRIPTION
**Pull request by emmenko**
Follow up of #651 to support passing intl values

### CODE REVIEW

File: src/components/typography/text/text.js [hunk 1]
```diff
@@ -186,6 +190,7 @@ Detail.propTypes = {
   title: nonEmptyString,
   truncate: PropTypes.bool,
   intlMessage: requiredIf(intlMessageShape, props => !props.children),
```
[the remark this answers is not in the document]
Comment (emmenko) [reply src/components/typography/text/text.js]: Hmm, I guess it makes sense yes. Thanks!

[the remark this answers is not in the document]
Comment (emmenko) [reply src/components/typography/text/text.js]: Hmm how would you write the condition then? 🤔

[the remark this answers is not in the document]
Comment (tdeekens) [reply src/components/typography/text/text.js]: `intlMessage: requiredIf(intlMessageShape, props => !props.children || props.intlValues),`?

[the remark this answers is not in the document]
Comment (emmenko) [reply src/components/typography/text/text.js]: Although, we might not need this prop after all.

We already spread the `intlMessage`, so we could simply add the `values` object into the `intlMessage`.

```js
<Text.Body intlMessage={{ ...messages.title, values: { count: 10 } }} />
```

[the remark this answers is not in the document]
Comment (tdeekens) [reply src/components/typography/text/text.js]: Yeah. Whatever API you find more convienent.

[the remark this answers is not in the document]
Comment (montezume) [reply src/components/typography/text/text.js]: Yeah I prefer that

File: src/components/typography/text/text.js [hunk 2]
```diff
@@ -49,6 +49,7 @@ Headline.propTypes = {
...
```
Comment (tdeekens) [inline src/components/typography/text/text.js]: BTW: This has caused breaking tests in the MC. Maybe it's an overly eager prop-type we could also remove entirely. As before you could have empty children. Now you can't, you have to have a `intlMessage` if no `children` are set which is a bit hard as a validation.

Comment (emmenko) [reply src/components/typography/text/text.js]: When would you have empty children? Does that even make sense? 🤔

Comment (tdeekens) [reply src/components/typography/text/text.js]: I am not saying it does make sense. I am saying we do have that and this was a breaking change as a result cause we allowed it before and then didn't.

Comment (emmenko) [reply src/components/typography/text/text.js]: I saw the PR now. I think it does not make sense to render the Text component with empty children. We should just not render the component in the first place.

```js
const title = undefined;

// wrong
<Text.Body>{title}</Text.Body>

// correct
{title && (<Text.Body>{title}</Text.Body>)}
```

WDYT?

**Comment by emmenko** [comment]
Alright, let's try with `intlMessage` only. Closing this PR then

File: src/components/typography/text/text.js (hunk 2 above)
Comment (tdeekens) [reply src/components/typography/text/text.js]: Yes it would make sense. However people didn't do that 🗡

**Pull request closed without merging**